books/spiders/book.py

```diff
@@ -0,0 +1,56 @@
+import scrapy
+from scrapy.http import Response
+
+
+class BookSpider(scrapy.Spider):
+    name = "book"
+    allowed_domains = ["books.toscrape.com"]
+    start_urls = ["https://books.toscrape.com/"]
+
+    def parse(self, response: Response, **kwargs) -> None:
+        for product_url in response.css("h3 > a::attr(href)").getall():
+            product_url = response.urljoin(product_url)
+            yield scrapy.Request(product_url, callback=self.parse_product)
+
+        next_page = response.css("li.next > a::attr(href)").get()
+
+        if next_page:
+            yield response.follow(next_page, callback=self.parse)
+
+    def extract_book_name(self, response: Response) -> str:
+        return response.css("h1::text").get()
+
+    def get_price(self, response: Response) -> int:
+        return response.css(".price_color::text").get()
```

Comment:
The `get_price` method returns a string representing the price. If you intend to return an integer or float, you should convert the string to the appropriate type, possibly after stripping currency symbols.